cplusplus/remove_duplicate_elements.cpp

```diff
@@ -1,34 +1,22 @@
 #include<iostream>
- 
 using namespace std;
  
 int main()
 {
 	int i,j,k,n,a[30];
-	cout<<"How many elements? ";
-	cin>>n;
+	cout<<"How many elements? "; cin>>n;
 	cout<<"\nEnter elements of array\n";
-	
 	for(i=0;i<n;i++)
 		cin>>a[i];
-		
 	for(i=0;i<n;i++)
 		for(j=i+1;j<n;)
-		{
-			if(a[i]==a[j])
-			{
+			if(a[i]==a[j]){
 				for(k=j;k<n-1;++k)
-					a[k]=a[k+1];
-					
-				--n;
+						a[k]=a[k+1];
+						--n;

```

Comment:
Can the --n be moved within the for loop so that that block is one line? Also, it looks like line 13's if and line 18's else can be compressed using `? :`

Reply:
i'm stuck :'(

Reply:
```c
#include <iostream>
using namespace std;

int main()
{
    int i, j, k, n;
    cout << "How many elements? "; cin >> n;
    int a[n];
    cout << endl << "Enter elements of array" << endl;
    for(i = 0;i < n; ++i) cin >> a[i];

    // One line insertion via == lookup loop
    for(i = 0; i < n; ++i) for(j = i + 1; j < n; ++j) if (a[i] == a[j]) for(--n, k = j--; k < n; ++k) a[k] = a[k + 1];

    cout << n << " unique" << endl;
    for(i = 0; i < n; ++i) cout << a[i] << ' ';
    cout << endl;
    return 0;
}
```
I believe I've come up with a solution that moves some stuff around, please have a try! I had to avoid using if/else because that would require a ; break, and I could not use ? : without redundancy in the for loop, so I stuck with just an if { for }.

One factor I'm trying to figure out is, what exactly was this assignment accomplishing:
```c
			if(a[i]==a[j]){
				for(k=j;k<n-1;++k)
						a[k]=a[k+1];
						```